netplugin/plugin/netplugin.go

```diff
@@ -303,27 +304,57 @@ func (p *NetPlugin) Reinit(cfg Config) {
 }
 
 //InitGlobalSettings initializes cluster-wide settings (e.g. fwd-mode)
-func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) {
+func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) error {
+
+	/*
+		Query global settings from state store
+		1. if forward mode or private net is empty, retry 1 sec
+		2. if forward mode doesn't match, return error
+		3. if stored private net is wrong, return error
+	*/
 
 	gCfg := mastercfg.GlobConfig{}
 	gCfg.StateDriver = stateDriver
-	err := gCfg.Read("")
-	if err != nil {
-		logrus.Errorf("Error reading forwarding mode from cluster store")
-		inst.FwdMode = "bridge"
-		inst.HostPvtNW = defaultPvtSubnet
-		return
+
+	// wait until able to get fwd mode and private subnet
+	for {
+		err := gCfg.Read("")
```

Comment:
If the error is _only_ checked immediately following the function call, it's Go convention to just check it inline in the if statement:

```go
if err := gCfg.Read(""); err != nil {
        logrus.Errorf("Error reading global settings from cluster store")
}
```

This style of assignment is also okay to use even if there is an existing error object in the current scope.  You can always use `:=` and the `err` var does not leak out into the function scope.

Reply:
done